src/tui/terminal.rs

```diff
@@ -83,9 +98,13 @@ pub fn enter() -> miette::Result<TerminalGuard> {
         .into_diagnostic()
         .wrap_err("Failed to create ratatui terminal")?;
 
+    INSIDE.store(true, Ordering::SeqCst);
```

Comment:
Hmm... I feel like `INSIDE` should be set ASAP rather than after we're finished doing all the `enter` tasks, so that an incomplete `enter` gets rolled back by `exit`.